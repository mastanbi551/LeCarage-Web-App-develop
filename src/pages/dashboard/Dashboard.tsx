import React, { useEffect, useState } from 'react';
import { Loader } from '../../components';
export const Dashboard = () => {
    const [loading, setLoading] = useState(true);

    useEffect(() => {
      setTimeout(() => {
        setLoading(false);
      }, 2000);
    }, []);

    return (
        <>
            {
                loading ? <Loader />
                :
                <div>Dashboard</div>
            }
        </>
    );
};