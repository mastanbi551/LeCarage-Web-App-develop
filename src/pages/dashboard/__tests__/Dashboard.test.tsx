import React from 'react';
import renderer from 'react-test-renderer';
import { Dashboard } from '../Dashboard';

describe('Dashboard component', () => {
  it('renders the Loader when loading is true', () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Dashboard content when loading is false', () => {
    const tree = renderer.create(<Dashboard />);

    setTimeout(() => {
      expect(tree.toJSON()).toMatchSnapshot();
    }, 2000);
  });
});
