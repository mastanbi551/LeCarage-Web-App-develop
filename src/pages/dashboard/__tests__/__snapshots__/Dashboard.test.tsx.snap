// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the Loader when loading is true 1`] = `
<div
  className="loader"
>
  <div
    aria-busy={true}
    aria-label="three-dots-loading"
    data-testid="three-dots-loading"
    role="status"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <svg
      data-testid="three-dots-svg"
      fill="#4fa94d"
      height="80"
      viewBox="0 0 120 30"
      width="80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="15"
        cy="15"
        r={15}
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
      <circle
        attributeName="fill-opacity"
        cx="60"
        cy="15"
        from="1"
        r={9}
        to="0.3"
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="9"
          repeatCount="indefinite"
          to="9"
          values="9;15;9"
        />
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="0.5"
          repeatCount="indefinite"
          to="0.5"
          values=".5;1;.5"
        />
      </circle>
      <circle
        cx="105"
        cy="15"
        r={15}
      >
        <animate
          attributeName="r"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="15"
          repeatCount="indefinite"
          to="15"
          values="15;9;15"
        />
        <animate
          attributeName="fill-opacity"
          begin="0s"
          calcMode="linear"
          dur="0.8s"
          from="1"
          repeatCount="indefinite"
          to="1"
          values="1;.5;1"
        />
      </circle>
    </svg>
  </div>
</div>
`;
